llm/orchestrator-server/src/main/python/app/src/llm_orchestrator/services/langchain/rag_chain.py

```diff
@@ -78,20 +83,26 @@ def execute_qa_chain(query: RagQuery, debug: bool) -> RagResponse:
         },
     )
 
+    logger.debug(
+        'RAG chain - Use chat history: %s', 'Yes' if len(query.history) > 0 else 'No'
+    )
     message_history = ChatMessageHistory()
     for msg in query.history:
         if ChatMessageType.HUMAN == msg.type:
             message_history.add_user_message(msg.text)
         else:
             message_history.add_ai_message(msg.text)
 
-    records_callback_handler = RetrieverJsonCallbackHandler()
-
     inputs = {
         **query.question_answering_prompt_inputs,
         'chat_history': message_history.messages,
     }
 
+    logger.debug(
+        'RAG chain - Use RetrieverJsonCallbackHandler for debugging : %s',
+        'Yes' if debug else 'No',
```

Comment:
```suggestion
        debug,
```

Without condition it will log True or False it's suffisant and prevent that condition for being executed even if debug logs are disabled